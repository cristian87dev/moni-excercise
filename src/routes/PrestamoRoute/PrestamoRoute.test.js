jest.unmock('redux-mock-store');

import React from 'react'; // eslint-disable-line
import configureStore from 'redux-mock-store'; // eslint-disable-line
import { Provider } from 'react-redux'; // eslint-disable-line
import thunk from 'redux-thunk'; // eslint-disable-line
import renderer from 'react-test-renderer'; // eslint-disable-line
import LoanRoute from './index'; // eslint-disable-line

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let store = null;

beforeEach(()=>{
  const initialState = { prestamo: { }, counter: { } };
  store = mockStore(initialState)
})

it('LoanRoute renders without crashing', () => {
  const component = renderer.create(
    <Provider store={store}>
      <LoanRoute />
    </Provider>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
