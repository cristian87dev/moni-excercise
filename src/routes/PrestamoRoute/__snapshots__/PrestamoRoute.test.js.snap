// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanRoute renders without crashing 1`] = `
<section
  className="section prestamo-route"
>
  <h2
    className="section-title"
  >
    Pedí Tu Prestamo
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="fields-group"
    >
      <div
        className="input-field"
      >
        <input
          autoComplete="off"
          className="input-value"
          id="Counter"
          name="Counter"
          onChange={[Function]}
          placeholder="Counter"
          type="text"
          value=""
        />
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Increment
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        Decrement
      </button>
    </div>
    <div
      className="fields-group"
    >
      <div
        className="input-field"
      >
        <input
          autoComplete="off"
          className="input-value"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field"
      >
        <input
          autoComplete="off"
          className="input-value"
          id="surname"
          name="surname"
          onChange={[Function]}
          placeholder="Apellido"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field"
      >
        <input
          autoComplete="off"
          className="input-value"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="fields-group"
    >
      <div
        className="input-field"
      >
        <select
          autoComplete="off"
          className="input-value"
          id="gender"
          name="gender"
          onChange={[Function]}
          placeholder="Género"
          value=""
        >
          <option
            value="M"
          >
            Masculino
          </option>
          <option
            value="F"
          >
            Femenino
          </option>
        </select>
      </div>
      <div
        className="input-field"
      >
        <input
          autoComplete="off"
          className="input-value"
          id="documentNumber"
          name="documentNumber"
          onChange={[Function]}
          placeholder="Número de Documento"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="fields-group"
    >
      <div
        className="input-field"
      >
        <input
          autoComplete="off"
          className="input-value"
          id="loanMount"
          name="loanMount"
          onChange={[Function]}
          placeholder="Valor del Prestamo"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="form-actions"
    >
      <button
        className="button submit"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
